dot/parachain/overseer/overseer.go

```diff
@@ -60,14 +71,18 @@ func NewOverseer(blockState BlockState) *Overseer {
 		blockState:           blockState,
 		activeLeaves:         make(map[common.Hash]uint32),
 		SubsystemsToOverseer: make(chan any),
-		subsystems:           make(map[Subsystem]chan any),
-		nameToSubsystem:      make(map[parachaintypes.SubSystemName]Subsystem),
+		subsystems:           make(map[parachaintypes.Subsystem]chan any),
+		nameToSubsystem:      make(map[parachaintypes.SubSystemName]parachaintypes.Subsystem),
 	}
 }
 
+func (o *Overseer) GetSubsystemToOverseerChannel() chan any {
```

Comment:
I will change this when I propose the service architecture I've mentioned before.  The producer of the channel will close this channel to notify the consumer (overseer in this case) that the service has halted.